scipy/stats/tests/test_qmc.py

```diff
@@ -396,10 +394,56 @@ def test_subclassing_QMCEngine():
     assert_equal(sample_2, sample_2_test)
     assert engine.num_generated == 12
 
+
+def test_raises():
```

Comment:
I see that some of these are tested elsewhere, but "Bounds are not consistent" is tested elsewhere but not here. I'd suggest testing consistently either in both places or just in one place, unless there's a reason not to.

Reply:
Done, I simplified the logic to only call once the input validation. This way it's actually external to this function.